### DISCUSSION TITLE
How to create a long function in the "validate" property

### DISCUSSION THREAD

**Discussion by Jsramosra** [react-hook-form/react-hook-form#8900] [Q&A]
**RHF Version:** "^7.28.0"
**React Version:** "^17.0.2"

Hello @bluebill1049 and React Hook form team.

**Problem Description**

I'm trying to build a custom validation within the "validate" property in the register method, which is a "long one", but unfortunately I was not able to make it work, this is the sample code that i was trying to run.

   ```
                     <TextField 
                            {...register("email1", {
                                validate:{
                                    containsNumber: v =>{
                                        var containsN = false;
                                        var i = 0;
                                        var char = "";
                                        while(i <= v.length){
                                            char = v.charAt(i);
                                            if(!isNaN(char * 1)) containsN = true;
                                            i++;
                                        }

                                        return containsN;
                                    }
                                }
                            ,required:true})}
                            error={Boolean(errors.email1)}
                            id="email1"
                            type="password"
                        />
```

**INPUT:** Test123-
**EXPECTED OUTPUT:** true
**CURRENT OUTPUT:** The custom fuction is not being triggered to be validated

Could you please advise the best way to do this.

Thanks!

**Comment by bluebill1049** [comment]
maybe you can turn this into a function and write a unit test against it, and see which part went wrong. Basically only `Boolean` value as `true` is consider a valid input.
